src/components/Checkbox/Checkbox.st.css

```diff
@@ -1,10 +1,103 @@
 :import {
-  -st-from: "wix-ui-core/index.st.css";
-  -st-named: Checkbox, CheckboxStyle;
+  -st-from: "wix-ui-core/dist/src/components/Checkbox/Checkbox.st.css";
+  -st-default: Checkbox;
+}
+
+:import {
+  -st-from: "../../palette.st.css";
+  -st-named: backgroundPrimary, backgroundSecondary, mainMutedBackground, mainMuted, main, mainHover, dangerMutedHover, danger, disabledFields, disabledCheck, dividerColor;
+}
+
+:vars {
+  medium: 16px;
 }
 
 .root {
   -st-extends: Checkbox;
-  -st-states: error, success, x-small, small;
-  -st-mixin: CheckboxStyle;
+  width: initial;
+  height: initial;
+  display: flex;
+  align-items: center;
+  outline: 0;
+}
+
+.root::box {
+  background: value(backgroundSecondary);
+  border: value(mainMuted);
+  border-radius: 4px;
+  border: solid 1px value(mainHover);
+  box-sizing: border-box;
+  width: value(medium);
+  height: value(medium);
+}
+
+.tickIcon {
+  width: 9px;
+  padding-left: 2.5px;
+  height: 9px;
+}
+
+.indeterminateIcon {
+  width: 8px;
+  padding-left: 3px;
+  height: 8px;
+}
+
+.root::childContainer {
+  font-size: 16px;
+  padding-left: 12px;
+}
+
+.root::box:active {
+  background-color: value(mainHover);
+}
+
+.root:focus-visible::box {
+  border-radius: 4px;
+  box-shadow: 0 0 0 3px #aadbfc;
+  border: solid 1px #3899ec;
+}
+
+.root:disabled::box {
+  background-color: value(backgroundPrimary);
+  border-color: value(disabledFields);
+}
+
+.root:disabled:checked::box {
+  border-color: value(dividerColor);
+}
+
+.root:disabled {
+  pointer-events: none;
+}
+
+.root:disabled::childContainer {
+  color: value(backgroundPrimary);
+}
+
+.root:error::box {
+  border-color: value(danger);
+}
+
+.root:hover:not(:disabled):not(:checked):not(:active)::box {
+  background-color: value(mainMutedBackground);
+}
+
+.root:checked:not(:active):not(:disabled)::box {
+  background-color: value(main);
+}
+
+.root::box > svg {
+  position: absolute;
+  top: 50%;
+  transform: translateY(-50%);
+  fill: value(backgroundSecondary);
+}
+
+.root:hover:indeterminate:not(:disabled):not(:checked):not(:active)::box > svg {
```

Comment:
Instead of adding `fill` to the SVG just add `color` to the box:

```
.root:hover:indeterminate:not(:disabled):not(:checked):not(:active)::box {
  color: value(mainMutedBackground);
}
```